### PR TITLE
Add mkdocs-linkcheck hook

### PR DESCRIPTION
**Pull request by totoroot**
Marking this as a draft as `mkdocs-linkcheck` is not yet in nixpkgs.
I'll probably package it this weekend and I'll link this draft to the PR in nixpkgs. I will convert this draft to a regular PR once the package gets merged into nixpkgs

### CODE REVIEW

File: modules/hooks.nix
Comment (totoroot) [inline modules/hooks.nix:1]: Not sure what has happened here. I only added lines.

Comment (totoroot) [reply modules/hooks.nix:1]: I force-pushed some changes to get a clean diff.

**Comment by totoroot** [comment]
@domenkozar `mkdocs-linkcheck` is now in nixpkgs, so this PR is ready for review and can get merged.

**Comment by domenkozar** [comment]
I'd like to point out that pre-commit hooks should be fast (max 1-2s), so that developers don't get frustrated with committing :)

**Comment by totoroot** [comment]
@domenkozar Yeah, I totally agree!
Admittedly `mkdocs-linkcheck` isn't fast, but since you opened [this issue for devenv](https://github.com/cachix/devenv/issues/50) yourself, I thought I'd add it as a pre-commit hook :grin:

**Pull request merged**